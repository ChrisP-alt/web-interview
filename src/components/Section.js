import React from 'react'
import PropTypes from 'prop-types'
import styles from '../styles/components/section.module.scss'

const Section = ({ children, headerText, Icon }) => (
  <div>
    <div className={styles.header}>
      <Icon />
      <h2>{headerText}</h2>
    </div>
    <div className={styles.content}>{children}</div>
  </div>
)

Section.propTypes = {
  children: PropTypes.node.isRequired,
  headerText: PropTypes.string.isRequired,
  Icon: PropTypes.func.isRequired,
}

export default Section
