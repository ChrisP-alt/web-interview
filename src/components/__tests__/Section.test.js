import React from 'react'
import { shallow } from 'enzyme'

import Section from '../Section'
import { ReactComponent as Icon } from '../../icons/comments-regular.svg'

describe('Section component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Section headerText="Section Header" Icon={Icon}>
        <div className="child" />
      </Section>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
