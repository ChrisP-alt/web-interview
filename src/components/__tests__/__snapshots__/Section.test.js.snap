// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <ReactComponent />
    <h2>
      Section Header
    </h2>
  </div>
  <div
    className="content"
  >
    <div
      className="child"
    />
  </div>
</div>
`;
